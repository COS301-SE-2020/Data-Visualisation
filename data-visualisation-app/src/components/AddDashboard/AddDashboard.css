
.add-dashboard-box {
    border: 1px solid #d6d6d6;
    box-shadow: 3px 3px 8px #dcdcdc;
    border-radius: 2%;
    padding: 30px;

    background-color: white;
    margin-top: 100px;
}

.add-dashboard-content {
    background-color: #c5c8c1;
}

/*h1{*/
/*    margin: auto;*/
/*    display: block;*/
/*    text-align: center;*/
/*    color: #d6883e;*/
/*}*/

/*#adderField{*/
/*    margin: auto;*/
/*    display: block;*/
/*}*/

/*#submitButton{*/
/*    background-color: #e7e7e7;*/
/*    color: black;*/
/*    border: none;*/
/*    padding: 15px 32px;*/
/*    text-align: center;*/
/*    text-decoration: none;*/
/*    display: inline-block;*/
/*    font-size: 16px;*/
/*    margin: 4px 2px;*/
/*    cursor: pointer;*/
/*    width: 20%;*/
/*}*/
/*#submitButton:hover{*/
/*    opacity:0.5;*/
/*    background-color: #d6883e;*/
/*}*/

/*form{*/
/*    position: relative;*/
/*    top: 120px;*/
/*    width: 50%;*/
/*    margin: auto;*/
/*    -webkit-box-shadow: 3px 3px 5px 6px #ccc;  */
/*    -moz-box-shadow:    3px 3px 5px 6px #ccc;  */
/*    box-shadow:         3px 3px 5px 6px #ccc;*/
/*    padding: 40px;*/
/*}*/
/*input{*/
/*    width: 100%;*/
/*    padding: 12px 20px;*/
/*    margin: 8px 0;*/
/*    display: inline-block;*/
/*    border: 1px solid #ccc;*/
/*    border-radius: 4px;*/
/*    box-sizing: border-box;*/
/*}*/
