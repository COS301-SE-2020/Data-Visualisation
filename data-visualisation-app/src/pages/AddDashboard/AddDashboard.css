
h1{
    margin: auto;
    display: block;
    text-align: center;
}

#adderField{
    margin: auto;
    display: block;
}

#submitButton{
    margin: auto;
    display: block;
}

form{
    position: relative;
    top: 120px;
    width: 50%;
    margin: auto;
    border: 5px solid #B14E6B;
    padding: 40px;
}