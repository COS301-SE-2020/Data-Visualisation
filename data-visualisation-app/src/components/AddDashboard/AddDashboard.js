import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import './AddDashboard.css';

function AddDashboard(props) {

    
    const [dashboardname, setDashboardname] = useState('');

      const mySubmitHandler = (event) => {
        //submit

        event.preventDefault();

        props.add({name:dashboardname});

        props.home();

      }
      

      const myChangeHandler = (event) => {
        setDashboardname(event.target.value);
      }


        return (
          <form onSubmit={mySubmitHandler}>
          <h1>Dashboard name:</h1>
          <input id = "adderField"
            type='text'
            onChange={myChangeHandler}
          />
          <input id = "submitButton"
            type='submit'
          />
          </form>
        );
      
    

  }

  export default AddDashboard;