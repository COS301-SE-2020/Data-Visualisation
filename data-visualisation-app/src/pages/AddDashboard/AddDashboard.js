import React, { useState } from 'react';
import './AddDashboard.css';

function AddDashboard(props) {
    const [dashBoardName, setDashboardName] = useState('');
    const [dashBoardDescription, setDashboardDescription] = useState('');
    const mySubmitHandler = (event) => {
        //submit
        event.preventDefault();
        props.add({name:dashBoardName, description:dashBoardDescription});
        props.home();
      };
      const myChangeHandler = (event) => {
          setDashboardName(event.target.value);
      };
      const myChangeHandler2 = (event) =>{
          setDashboardDescription(event.target.value);
      };
        return (
          <form onSubmit={mySubmitHandler}>
            <div>
              <h1>Dashboard name:</h1>
              <br/>
              <input
                id = "adderField"
                data-testid = "add-field"
                type='text'
                placeholder = "Consumer"
                value= {dashBoardName}
                onChange={myChangeHandler}
              /><br/>
              <h1>Description:</h1>
              <input
                  id = "adderField1"
                  data-testid = "add-field1"
                  type='text'
                  placeholder = "Some information about dashboard"
                  value= {dashBoardDescription}
                  onChange={myChangeHandler2}
              /><br/>
              <br/>
              <input
                id = "submitButton" 
                data-testid = "add-button"
                type='submit'
              />
              </div>
          </form>
        );
      
    

  }

  export default AddDashboard;