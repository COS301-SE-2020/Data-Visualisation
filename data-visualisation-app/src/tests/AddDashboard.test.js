import React from 'react';
import AddDashboard from '../components/AddDashboard';
import renderer from 'react-test-renderer';


Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
    })),
});

describe('AddDashboard component', () => {
    it('AddDashboard renders correctly', () => {
        const rendered = renderer.create(<AddDashboard/>);
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});

