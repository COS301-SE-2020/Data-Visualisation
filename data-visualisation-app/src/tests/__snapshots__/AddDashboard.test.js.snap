// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDashboard component AddDashboard renders correctly 1`] = `
<main
  className="ant-layout-content"
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <div
    className="ant-row"
    style={
      Object {
        "marginBottom": 8,
        "marginLeft": -8,
        "marginRight": -8,
        "marginTop": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-8 ant-col-offset-8"
      style={
        Object {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        }
      }
    >
      <div
        className="add-dashboard-box"
      >
        <h4
          className="ant-typography"
          style={
            Object {
              "WebkitLineClamp": null,
              "marginBottom": "30px",
            }
          }
        >
          Create New Dashboard
        </h4>
        <form
          className="ant-form ant-form-horizontal"
          id="basic"
          onSubmit={[Function]}
        >
          <div
            className="ant-row ant-form-item"
            onChange={[Function]}
            placeholder="Consumer"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className="ant-form-item-required"
                htmlFor="basic_adderField"
                title="Name:"
              >
                Name
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    className="ant-input"
                    id="basic_adderField"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            onChange={[Function]}
            placeholder="Short Description"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className="ant-form-item-required"
                htmlFor="basic_adderField1"
                title="Description:"
              >
                Description
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <textarea
                    className="ant-input"
                    id="basic_adderField1"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    style={Object {}}
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-32 ant-col-offset-8 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <div
                    className="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      className="ant-space-item"
                      style={
                        Object {
                          "marginRight": 10,
                        }
                      }
                    >
                      <button
                        className="ant-btn ant-btn-primary"
                        onClick={[Function]}
                        type="submit"
                      >
                        <span>
                          Create
                        </span>
                      </button>
                    </div>
                    <div
                      className="ant-space-item"
                      style={Object {}}
                    >
                      <button
                        className="ant-btn"
                        onClick={[Function]}
                        type="button"
                      >
                        <span>
                          Cancel
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</main>
`;
